x/migration/keeper/msg_server_claim_morse_application.go

```diff
@@ -24,8 +24,55 @@ import (
 func (k msgServer) ClaimMorseApplication(ctx context.Context, msg *migrationtypes.MsgClaimMorseApplication) (*migrationtypes.MsgClaimMorseApplicationResponse, error) {
 	sdkCtx := cosmostypes.UnwrapSDKContext(ctx)
 	logger := k.Logger().With("method", "ClaimMorseApplication")
+	waiveMorseClaimGasFees := k.GetParams(sdkCtx).WaiveMorseClaimGasFees
 
-	if err := msg.ValidateBasic(); err != nil {
+	// Ensure that gas fees are NOT waived if one of the following is true:
```

Comment:
I don't fully understand why we are why are we adding this logic in the PR? 

If it should leverage `morseAccountClaimingIsEnabled`, should it be included in the `deer` in some way?

Reply:
Apologies for the confusion. :sweat_smile: 

_TL;DR, we should be enforcing gas fee waiver rules on all migration message handlers._

This is unrelated to the feature flag, but was previously omitted (only included in the Morse account claim handler). I was correcting this unintentional omission.